src/app/modules/recipe-header/containers/recipe-header.component.ts

```diff
@@ -24,4 +27,20 @@ export class RecipeHeaderComponent  {
   login(): void {
     this.router.navigate(['/account/signin'], { queryParams: { returnUrl: this.router.url } });
   }
+
+  public onSearch(search: string | null): void {
+    this.router.navigate(["./"], {
+      queryParams: { search: search, page: 1 },
+      queryParamsHandling: 'merge'
+    })
+  }
+
+  public onReset(): void {
+    var snapshot = this.activeRoute.snapshot;
+    const params = { ...snapshot.queryParams };
+    delete params.search
+    this.router.navigate([], {
+      queryParams: params
+    });
+  }
```

Comment:
did you try to do this: 
```suggestion
  public onReset(): void {
    this.router.navigate([], {
      queryParams: {search: null}, // or undefined
       queryParamsHandling: 'merge'
    });
  }
```